File: work/DSA/c++/recursion_1b/removeX.cpp
#include <bits/stdc++.h>
using namespace std;

void shiftCharacters(char input[])
{
    for (int i = 0; i < strlen(input); i++)
    {
        input[i] = input[i + 1];
    }
}

void removeX(char input[])
{
    if (input[0] == '\0')
    {
        return;
    }
    removeX(input + 1);
    if (input[0] == 'x')
    {
        shiftCharacters(input);
    }
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    char input[100];
    cin.getline(input, 100);
    removeX(input);
    cout << input << endl;
}

File: work/DSA/javascript/recursion_1b/removeX.js
let input = ["x", "a", "x", "b"]

function shiftCharacters(input, currentIndex) {
    for (let i = currentIndex; i < input.length - 1; i++) {
        input[i] = input[i + 1]
    }
    delete input[input.length - 1]
}

function removeX(input, currentIndex) {
    if (input[currentIndex] == undefined) {
        return
    }
    removeX(input, currentIndex + 1)
    if (input[currentIndex] == 'x') {
        shiftCharacters(input, currentIndex)
    }
}

removeX(input, 0)
for (i of input) {
    console.log(i)
}